From 6ee39aa2ed730859d9eb6f84fe84802dbe9933b3 Mon Sep 17 00:00:00 2001
From: yaozenhu <yaozenhu@realtek.com>
Date: Thu, 20 Oct 2016 14:01:50 +0800
Subject: [PATCH 1/3] [fix] RDK-117 [root cause] no audio data is decoded on AV
 offset problem stream to cause gstreamer can't complete preroll state
 [solution] increase multiqueue buffer let audio data can be send to audio
 decoder to be decoded [reviewer] Paul

Change-Id: I35cc1ae829a4e0448cd8f495d8b1be88b49610bc
---
 gst/playback/gstdecodebin2.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gst/playback/gstdecodebin2.c b/gst/playback/gstdecodebin2.c
index 7eb89da..e517953 100644
--- a/gst/playback/gstdecodebin2.c
+++ b/gst/playback/gstdecodebin2.c
@@ -240,8 +240,8 @@ enum
 
 /* when playing, keep a max of 2MB of data but try to keep the number of buffers
  * as low as possible (try to aim for 5 buffers) */
-#define AUTO_PLAY_SIZE_BYTES        2 * 1024 * 1024
-#define AUTO_PLAY_SIZE_BUFFERS      5
+#define AUTO_PLAY_SIZE_BYTES        10 * 1024 * 1024
+#define AUTO_PLAY_SIZE_BUFFERS      500
 #define AUTO_PLAY_SIZE_TIME         0
 
 #define DEFAULT_SUBTITLE_ENCODING NULL
@@ -3740,8 +3740,8 @@ decodebin_set_queue_size_full (GstDecodeBin * dbin, GstElement * multiqueue,
   GST_DEBUG_OBJECT (multiqueue, "setting limits %u bytes, %u buffers, "
       "%" G_GUINT64_FORMAT " time", max_bytes, max_buffers, max_time);
   g_object_set (multiqueue,
-      "max-size-bytes", max_bytes, "max-size-time", max_time,
-      "max-size-buffers", max_buffers, NULL);
+      "max-size-bytes", AUTO_PLAY_SIZE_BYTES, "max-size-time", max_time,
+      "max-size-buffers", AUTO_PLAY_SIZE_BUFFERS, NULL);
 }
 
 /* gst_decode_group_new:
-- 
2.7.4

